Hazards

joshuajgit opened the pull request:
Ready to move hazards over to the rest of the project

joshuajgit committed:
stage hazards

Moved the stage hazards over to /dev branch

joshuajgit committed:
quick fixes

- resized hazards to be the same size as level tiles
- added scene for hazards testing

Megan-VanHumbeck commented on Assets/CodeBase/DestroyThis.cs:
```diff
@@ -0,0 +1,16 @@
+/// <summary>
+/// Script to destroy object it is attached to after a specified delay in seconds.
+/// </summary>
+public class DestroyThis : MonoBehaviour
```
I don't really understand the necessity of this script. Could you please explain?

DavidoBod replied:
If it's to make the delay time universal (before deleting an object) I'd say just scrap it and input the delay manually in each class where those objects are being destroyed. But let me know if I'm missing the point of this class

DavidoBod commented on Assets/CodeBase/Entities/Hazards/Landmine.cs:
```diff
@@ -0,0 +1,31 @@
+        GameObject e = Instantiate(explosion, transform);
+        e.transform.localPosition = Vector3.zero;
+        e.transform.parent = null;
+        Destroy(this.gameObject);
```
We should kill the player too when they step on the landmine

JesseDes replied:
The explosion that it adds kills the player

JesseDes committed:
Restored Scene1

Changed size of explosion to make it feel more dodgeable and added trigger so explosion kills the player
-restored scene1
